wellcomeml/datasets/hoc.py

```diff
@@ -0,0 +1,41 @@
+import random
+import csv
+import os
+
+from wellcomeml.datasets.download import check_cache_and_download
+
+def load_split(data_path):
+    X = []
+    Y = []
+    with open(data_path) as f:
+        csvreader = csv.DictReader(f, delimiter='\t')
+        for line in csvreader:
+            X.append(line["sentence"])
+            Y.append(line["labels"].split(','))
+    return X, Y
+
+def load_hoc(split='train', shuffle=True):
+    path = check_cache_and_download("hoc")
+
+    train_data_path = os.path.join(path, "train.tsv")
+    X_train, Y_train = load_split(train_data_path)
+
+    test_data_path = os.path.join(path, "test.tsv")
+    X_test, Y_test = load_split(test_data_path)
+
+    if split == 'train':
+        X = X_train
+        Y = Y_train
```

Comment:
**suggestion**: I guess you are unnecessarily loading and splitting the test data in this case (and vice versa for split='test') consider only loading when it's needed

Reply:
+1 for that suggestion.

Reply:
it was done to avoid repetition of code but will revert